### PR TITLE
add blockchain.Stop as defer to test

### LINKED ISSUE #620: Review `core/blockchain_test.go` tests for appropriate use of blockchain.Stop
In reviewing the geth update for coreth, @ceyonur points out
> did we specifically skip defer chain.Stop()s after NewBlockchain calls?

We should review this file to ensure we don't leak unnecessary goroutines when possible in tests.

### PR DESCRIPTION
**Pull request by najeal**
## Why this should be merged

It resolves and close #620 
All tests run from `blockchain_test.go` had `defer blockchain.Stop()` instruction except one.

## How this works

One test has been changed to close the blockchain from the defer instruction if test fails.

### CODE REVIEW

File: core/test_blockchain.go [hunk 1]
```diff
@@ -1003,17 +1003,20 @@ func TestReorgReInsert(t *testing.T, create func(db ethdb.Database, chainConfig
 	checkBlockChainState(t, blockchain, gspec, chainDB, create, checkState)
 }
 
-//nolint:goimports
 // Insert two different chains that result in the identical state root.
 // Once we accept one of the chains, we insert and accept A3 on top of the shared
 // state root
-//   G   (genesis)
-//  / \
+//
```
Comment (aaronbuchwald) [inline core/test_blockchain.go:1009]: nit: revert unrelated comment changes

File: core/test_blockchain.go [hunk 2]
```diff
@@ -1187,6 +1193,16 @@ func TestReprocessAcceptBlockIdenticalStateRoot(t *testing.T, create func(db eth
...
+	blockchainToStop := blockchain
+	alreadyStoped := false
+	blockchainStopOnlyOnce := func() {
+		if alreadyStoped {
+			return
+		}
+		alreadyStoped = true
+		blockchainToStop.Stop()
+	}
+	defer blockchainStopOnlyOnce()
```
Comment (aaronbuchwald) [inline core/test_blockchain.go:1205]: Could we change this to use `sync.Once` so that it's using an easy to understand primitive? Maybe something like this?

```suggestion
	var stopOnce sync.Once
	defer stopOnce.Do(blockchain.Stop())
	.
	.
	.
	stopOnce.Do(blockchain.Stop())
```

**Review by aaronbuchwald — CHANGES_REQUESTED** [review]
Thanks for putting up this PR and your contribution!

Left a few comments

File: core/test_blockchain.go (hunk 2 above)
Comment (najeal) [reply core/test_blockchain.go]: done

File: core/test_blockchain.go (hunk 1 above)
Comment (najeal) [reply core/test_blockchain.go:1011]: it may come from the code editor. I reverted 👍

Comment (aaronbuchwald) [reply core/test_blockchain.go:1011]: Thanks the linter can be really aggressive with these diagrams and we haven't found a good setting that keeps them easy to read and satisfies the linter.

If you're interested in learning/contributing more, it would be pretty easy to review and be good exposure to how the VM/engine interact. Documentation for the invariants between them is here: https://github.com/ava-labs/avalanchego/blob/master/vms/README.md

Comment (aaronbuchwald) [reply core/test_blockchain.go:1011]: Could you remove the added empty new lines, so that there is no unrelated diff as part of this PR?

Comment (najeal) [reply core/test_blockchain.go]: @aaronbuchwald Thank you for the link ! I start slowly to look at some easy issues on `avalanche` `subnet-evm` and `avalanche-cli`.
If you have some issues for new contributors don't hesitate to ping me.

**Review by aaronbuchwald — APPROVED** [review]
LGTM - thanks for contributing!

**Review by darioush — APPROVED** [review]

File: core/test_blockchain.go (hunk 1 above)
Comment (aaronbuchwald) [reply core/test_blockchain.go]: What's the best place to ping you?

**Comment by aaronbuchwald** [comment]
Hey @najeal , I didn't notice this before, but could you make sure all of the commits on this PR are signed?

Full contributing guidelines can be found here: https://github.com/ava-labs/subnet-evm/blob/master/.github/CONTRIBUTING.md

File: core/test_blockchain.go (hunk 1 above)
Comment (najeal) [reply core/test_blockchain.go]: You can identify me on some issues, I should receive the notifications.
Otherwise if you have a valid discord invitation to Avalanche server it would be a good way too
